File: examples/10-benchmark.lua
require('os')
function bench_1()
    local start_time = os.clock()
    local round = 1000000
    local i = 1
    while true do
        local arr = {}
        local j
        for j = 0, round do
            arr[j] = j * j
        end
        print((os.clock() - start_time) / i / round)
        i = i + 1
    end
end
function bench_2()
    local start_time = os.clock()
    local round = 1000000
    local i = 1
    while true do
        local arr = {}
        local obj = {['foo'] = {{}, {}}}
        for j = 0, round do
            obj['foo'][1]['bar'] = j
            obj['foo'][2][j] = 'Hello,' .. 'World!'
        end
        print((os.clock() - start_time) / i / round)
        i = i + 1
    end
end
bench_2()

File: examples/10-benchmark.js
function bench_1() {
    let start_time = clock();
    let round = 1000000;
    for (let i = 1;; i++) {
        gc();
        let arr = [];
        for (let j = 0; j < round; j++) {
            arr[j] = j * j;
        }
        print((clock() - start_time) / round / i);
    }
}
function bench_2() {
    let start_time = clock();
    let round = 1000000;
    for (let i = 1;; i++) {
        gc();
        let obj = {"foo" : [ {}, [] ]};
        for (let j = 0; j < round; j++) {
            obj.foo[0].bar = j;
            obj.foo[1][j] = "Hello," + "World!";
        }
        print((clock() - start_time) / round / i);
    }
}
bench_2();

